### ISSUE TITLE
FIX Examples

### ISSUE THREAD

**Issue by kadel** [kedgeproject/kedge#483]
email from Burr:

3 for 3 fails with my first attempt.  Here are my notes:

likely dumb user error :-)  Also, I am trying to use minishift, perhaps it just does not work for minishift.

Try Kedge on Minishift v1.9.0+a511b25

started minishift with 
MINISHIFT_ENABLE_EXPERIMENT=y minishift start --profile minishift-1.9.0 --metrics --cpus 3 --memory 8192 --vm-driver virtualbox --openshift-version v3.7.0-rc.0

Download kedge

curl -L https://github.com/kedgeproject/kedge/releases/download/v0.5.0/kedge-darwin-amd64 -o kedge

chmod +x kedge

./kedge

Try Guestbook

https://github.com/kedgeproject/kedge/tree/master/examples/guestbook-demo

note: 
 - did not tell me to make a namespace
 - did not tell me to login
 - did not tell me to insure kubectl was in my path
 - where did the quay.io/tomkral image come from? is this on dockerhub?

./kedge create -f kedge_project/examples/guestbook-demo/backend.yaml -f kedge_project/examples/guestbook-demo/frontend.yaml -f kedge_project/examples/guestbook-demo/db.yaml

note: 
- bad news, damn, in the wrong namespace
- good news kedge delete seems to work!
oc expose service frontend

Fail
- mongo pod went crashloopbackoff crazy
INFO  ==> Starting mongodb... 
sed: can't read /bitnami/mongodb/conf/mongodb.conf: No such file or directory
INFO  ==> Starting mongod...
error: failed switching to "mongo": unable to find user mongo: no matching entries in passwd file

Try Wordpress
https://github.com/kedgeproject/kedge/tree/master/examples/wordpress

oc new-project kedge-wordpress
kedge create -f kedge_project/examples/wordpress/wordpress.yaml -f kedge_project/examples/wordpress/mariadb.yaml

Fail
- wordpress pod went crashlookbackoff crazy
tar: ./wp-signup.php: Cannot open: Permission denied
tar: .: Cannot utime: Operation not permitted
tar: Exiting with failure status due to previous errors

Try Gitlab
https://github.com/kedgeproject/kedge/tree/master/examples/gitlab

oc new-project kedge-gitlab

kedge create -f kedge_project/examples/gitlab/gitlab.yml -f kedge_project/examples/gitlab/redis.yml -f kedge_project/examples/gitlab/postgres.yml 

Fail
- All pods fail

**Comment by surajssd** [comment]
The way I see at this problem exists because we have tried those examples on minikube or minishift with addons like `anyuid` and `admin` user enabled. Now how do we solve this problem as kedge devs?

- Should we go and fix the images (which is lot of work, which also involves maintaining those images). But this will make our examples run anywhere no matter what, without user having to do any extra steps.
- Document what addons to enable so these examples work without problem there, but then that will work on OpenShift in minishift, in environment like OpenShift online this might not work.

**Comment by kadel** [comment]
Ideally, we should make sure that all our examples use images that don't require root

**Comment by surajnarwade** [comment]
As of Now, I think updating README with respective example is enough I guess

**Comment by kadel** [comment]
> The way I see at this problem exists because we have tried those examples on minikube or minishift with addons like anyuid and admin user enabled. Now how do we solve this problem as kedge devs

simple, newer use anyuid on openshift ;-)

**Comment by kadel** [comment]
I've fixed guestbook-demo example - https://github.com/kedgeproject/kedge/pull/484

**Comment by ashetty1** [comment]
Also, this issue is somewhat related to #376 ?

**Comment by cdrage** [comment]
Yup, let's get #376 up and running so we don't run into this issue in the future :+1:

**Comment by cdrage** [comment]
https://github.com/kedgeproject/kedge/issues/376 is almost complete. I've ran into errors regarding OpenShift tests which I've remarked here: #568
